pkg/query/selection.go

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package query
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+// Operator is a query operator
+type Operator string
+
+const (
+	// EqualsOperator takes two operands and tests if they are equal
+	EqualsOperator Operator = "="
+	// NotEqualsOperator takes two operands and tests if they are not equal
+	NotEqualsOperator Operator = "!="
+	// GreaterThanOperator takes two operands and tests if the left is greater than the right
+	GreaterThanOperator Operator = "gt"
+	// LessThanOperator takes two operands and tests if the left is lesser than the right
+	LessThanOperator Operator = "lt"
+	// InOperator takes two operands and tests if the left is contained in the right
+	InOperator Operator = "in"
+	// NotInOperator takes two operands and tests if the left is not contained in the right
+	NotInOperator Operator = "notin"
+	// EqualsOrNilOperator takes two operands and tests if the left is equal to the right, or if the left is nil
+	EqualsOrNilOperator Operator = "eqornil"
+)
+
+// IsMultiVariate returns true if the operator requires right operand with multiple values
+func (op Operator) IsMultiVariate() bool {
+	return op == InOperator || op == NotInOperator
+}
+
+// IsNullable returns true if the operator can check if the left operand is nil
+func (op Operator) IsNullable() bool {
+	return op == EqualsOrNilOperator
```

Comment:
the nil part will be ignored, so it will have the same effect as EqualsOperator. I don't think this is enough of a reason to separate into different operators

Reply:
eqornil and eq would work similarly for labels but its documented that they work differently which is confusing, someone might expect that eqornil works as specified since it can be used for labels - i dont insist on separating but it just looks cleaner and less confusing (and its just the operators and not the querying logic itself)